mhkit/dolfyn/io/rdi.py

```diff
@@ -0,0 +1,1096 @@
+            self.f.read_ui8(4 * self.cfg['n_cells'])
+        ).reshape((self.cfg['n_cells'], 4))
+        self._nbyte = 2 + 4 * self.cfg['n_cells']
+
+    def read_status(self,):
+        self.vars_read += ['status']
+        self.ensemble.status[:, :, self.ensemble.k] = np.array(
+            self.f.read_ui8(4 * self.cfg['n_cells'])
+        ).reshape((self.cfg['n_cells'], 4))
+        self._nbyte = 2 + 4 * self.cfg['n_cells']
+
+    def read_bottom(self,):
+        self.vars_read += ['dist_bt', 'vel_bt', 'corr_bt', 'amp_bt',
+                           'prcnt_gd_bt']
+        fd = self.f
+        ens = self.ensemble
+        k = ens.k
+        cfg = self.cfg
+        if self._source == 2:
+            print('Warning: lat/lon bottom reading passed')
+            # self.vars_read += ['latitude_gps', 'longitude_gps']
+            # fd.seek(2, 1)
+            # long1 = fd.read_ui16(1)
+            # fd.seek(6, 1)
+            # ens.latitude_gps[k] = fd.read_i32(1) * self._cfac
+            # if ens.latitude_gps[k] == 0:
+            #     ens.latitude_gps[k] = np.NaN
+        else:
+            fd.seek(14, 1)
+        ens.dist_bt[:, k] = fd.read_ui16(4) * 0.01
+        ens.vel_bt[:, k] = fd.read_i16(4) * 0.001
+        ens.corr_bt[:, k] = fd.read_ui8(4)
+        ens.amp_bt[:, k] = fd.read_ui8(4)
+        ens.prcnt_gd_bt[:, k] = fd.read_ui8(4)
+        if self._source == 2:
+            print('Warning: lat/lon bottom reading passed')
+            # fd.seek(2, 1)
+            # ens.longitude_gps[k] = (long1 + 65536 * fd.read_ui16(1)) * self._cfac
+            # if ens.longitude_gps[k] > 180:
+            #     ens.longitude_gps[k] = ens.longitude_gps[k] - 360
+            # if ens.longitude_gps[k] == 0:
+            #     ens.longitude_gps[k] = np.NaN
+            # fd.seek(16, 1)
+            # qual = fd.read_ui8(1)
+            # if qual == 0:
+            #     print('  qual==%d,%f %f' % (qual,
+            #                                 ens.latitude_gps[k],
+            #                                 ens.longitude_gps[k]))
+            #     ens.latitude_gps[k] = np.NaN
+            #     ens.longitude_gps[k] = np.NaN
+            # fd.seek(71 - 45 - 16 - 17, 1)
+            # self._nbyte = 2 + 68
+        else:
+            fd.seek(71 - 45, 1)
+            self._nbyte = 2 + 68
+        if cfg['prog_ver'] >= 5.3:
+            fd.seek(78 - 71, 1)
+            ens.dist_bt[:, k] = ens.dist_bt[:, k] + fd.read_ui8(4) * 655.36
+            self._nbyte += 11
+            #!!! Assumption: never running bottom-tracking without WinRiver or VMDAS
+            # if cfg['name'] == 'wh-adcp':
+            #     if cfg['prog_ver'] >= 16.20:
+            #         # RDI documentation claims these extra bytes were added in
+            #         # v8.17, but they don't appear in my 8.33 data -
+            #         # conversation with Egil suggests they were added in 16.20
+            #         fd.seek(4, 1)
+            #         self._nbyte += 4
```

Comment:
Is this assumption made by commenting out the code below or when the code below is active? Do we need this code?